src/client/actions/MessageCreate.js

```diff
@@ -15,16 +15,30 @@ class MessageCreateAction extends Action {
           messages[i] = channel._cacheMessage(new Message(channel, data[i], client));
         }
         channel.lastMessageID = messages[messages.length - 1].id;
-        if (user) user.lastMessageID = messages[messages.length - 1].id;
-        if (member) member.lastMessageID = messages[messages.length - 1].id;
+        channel.lastMessage = messages[messages.length - 1];
+        if (user) {
+          user.lastMessageID = messages[messages.length - 1].id;
+          user.lastMessage = messages[messages.length - 1];
+        }
+        if (member) {
+          member.lastMessageID = messages[messages.length - 1].id;
+          member.lastMessage = messages[messages.length - 1];
+        }
```

Comment:
Could you make this code more DRY? This code feels awkward because it indexes the messages array like 6 times.